src/ns-support.c

```diff
@@ -277,6 +277,36 @@ struct sc_ns_group *sc_open_ns_group(const char *group_name)
 	return group;
 }
 
+struct sc_ns_group *sc_maybe_open_ns_group(const char *group_name)
+{
+	struct sc_ns_group *group = sc_alloc_ns_group();
+	debug("opening namespace group directory %s", sc_ns_dir);
+	group->dir_fd =
+	    open(sc_ns_dir, O_DIRECTORY | O_PATH | O_CLOEXEC | O_NOFOLLOW);
+	if (group->dir_fd < 0) {
+		if (errno == ENOENT) {
+			free(group);
+			return NULL;
+		}
```

Comment:
This function is identical to sc_open_ns_group() except for this ENOENT check. It would be more maintainable to instead modify sc_open_ns_group() to take a flag and decide how to error based on the flag. Eg:

```
struct sc_ns_group *sc_open_ns_group(const char *group_name, const int graceful_enoent)
{
...
	if (group->dir_fd < 0) {
                if (graceful_enoent == 1 && errno == ENOENT) {
                        free(group);
                        return NULL;
                }
                die("cannot open directory for namespace group %s", group_name);
        }
...
```

Then sc-main.c uses:
```
group = sc_open_ns_group(group_name, 0);
```

and snap-discard-ns.c uses:
```
group = sc_open_ns_group(group_name, 1);
```

Obviously feel free to rename graceful_enoent and be sure to document this argument.